python/cudf/cudf/core/column/struct.py

```diff
@@ -71,6 +71,21 @@ def copy(self, deep=True):
             result = result._rename_fields(self.dtype.fields.keys())
         return result
 
+    def as_interval_column(self, closed="right"):
+        from cudf.core.column import IntervalColumn
+
+        return IntervalColumn(
+            size=self.size,
+            dtype=cudf.core.dtypes.IntervalDtype(
+                self.dtype.fields["left"], closed
+            ),
+            mask=self.mask,
+            offset=self.offset,
+            null_count=self.null_count,
+            children=self.children,
+            closed=closed,
+        )
```

Comment:
Why do we need this method? Instead of this method I would imagine just calling `IntervalColumn.from_struct_column(my_struct_column)` for example.

Reply:
This is kinda a circular dependency currently.

Reply:
All of our `astype()` code is based on column classes having `as_x_column()` defined. So ideally we need an `as_interval_column()` for any type we expect to convert to an `IntervalColumn`.

Reply:
I'm fine to leave this for now and fix in a follow up, but we're creating circular dependencies this way. `StructColumn` would depend on `IntervalColumn`, while `IntervalColumn` inherits from `StructColumn`.

In our other cases, the columns aren't dependent on one another otherwise, so it's fine, but this is a case where there's already a logical dependency.